judge/management/commands/generate_report.py

```diff
@@ -0,0 +1,97 @@
+            else:
+                row += '<td></td>\n'
+        row += '<td>%.0f</td>\n' % participation.score
+
+        out += '<tr>\n'
+        out += '<th>%s</th>\n' % registrant.user.user.get_full_name()
+        out += row
+        out += '</tr>\n'
+    return out
+
+
+class Command(BaseCommand):
+    help = 'generates a report for an ICS3U contest'
+
+    def add_arguments(self, parser):
+        parser.add_argument('key', help='key of the contest to generate a report for')
+        parser.add_argument('--period', help='ics3u class period of the contest', type=int)
+        parser.add_argument('--dry-run', help='don\'t actually email the report', action='store_true')
+        parser.add_argument('--directory', help='specify where to store the report')
+        parser.add_argument('-p', '--phantomjs', action='store_const', const=PhantomJSPdfMaker,
+                            default=DefaultPdfMaker, dest='engine')
+        parser.add_argument('-s', '--slimerjs', action='store_const', const=SlimerJSPdfMaker, dest='engine')
+        parser.add_argument('-c', '--chrome', '--puppeteer', action='store_const',
+                            const=PuppeteerPDFRender, dest='engine')
+        parser.add_argument('-S', '--selenium', action='store_const', const=SeleniumPDFRender, dest='engine')
+
+    def handle(self, *args, **options):
+        try:
+            contest = Contest.objects.get(key=options['key'])
+        except Contest.DoesNotExist:
+            print('Bad contest code')
+            return
+
+        directory = options['directory']
+        scoreboard = generate_scoreboard(contest, options['period'])
+        if options['period'] is not None:
+            teacher = settings.DMOJ_ICS_REPORT_PERIODS[options['period'] - 1]
+        else:
+            teacher = None
+        with options['engine'](directory, clean_up=directory is None) as maker:
+            maker.html = get_template('contest/ics3u_report.html').render({
+                'contest': contest,
+                'contest_problems': contest.contest_problems.all(),
+                'scoreboard': scoreboard,
+                'teacher': teacher,
+                'math_engine': maker.math_engine,
+            }).replace('"//', '"https://').replace("'//", "'https://")
+            maker.template['footerTemplate'] = "<div></div>"
+            maker.title = contest.name
+            for file in ('style.css', 'pygment-github.css', 'mathjax_config.js'):
+                maker.load(file, os.path.join(settings.DMOJ_RESOURCES, file))
+            maker.make(debug=True)
+            if not maker.success:
+                print(maker.log, file=sys.stderr)
+            elif directory is None:
+                shutil.move(maker.pdffile, contest.key + '.pdf')
+
+            if options['period'] is not None and not options['dry_run']:
+                email = EmailMessage(
+                    'ICS3U Contest Report',
+                    f'Dear {teacher[0]},\n\nHere are the results, '
```

Comment:
Change to the following:
Subject: `{{ contest.name }} Report`
Body:
```
Dear {{ teacher }},

Attached are the results, problems, and editorials for the {{ contest.name }}.
```

Also, @6167656e74323431 should we mention that this is an (semi-)automated email and to contact us if there's an issue?

Reply:
Yes.